iceoryx_binding_c/include/iceoryx_binding_c/wait_set.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_BINDING_C_WAIT_SET_H
+#define IOX_BINDING_C_WAIT_SET_H
+
+#include "iceoryx_binding_c/condition.h"
+#include "iceoryx_binding_c/enums.h"
+#include "iceoryx_binding_c/internal/c2cpp_binding.h"
+#include "iceoryx_binding_c/subscriber.h"
+#include "iceoryx_binding_c/types.h"
+
+#include <time.h>
+
+/// @brief wait set handle
+typedef CLASS WaitSet* iox_wait_set_t;
+
+/// @brief initialize wait set handle
+/// @param[in] self pointer to preallocated memory of size = sizeof(iox_wait_set_storage_t)
+/// @return handle to wait set
+iox_wait_set_t iox_wait_set_init(iox_wait_set_storage_t* self);
```

Comment:
personally I would vote for `iox_ws_init` the `_wait_set_` distracts to much from the actual thing the fuction does, especially with `iox_wait_set_detach_condition`. Does this set a detach condition?

Reply:
You are right - solved!